lib/WWW/YouTube/Download.pm

```diff
@@ -442,6 +442,216 @@ sub user_id {
+	){
+		my $next = $playlistVideoListRenderer->{continuations}->[0]->{nextContinuationData};
+		my $continuation = $next->{continuation};
+		my $ctoken = $continuation;
+		my $itct = $next->{clickTrackingParams};
+
+		if($self->{verbose}){
+			print " next page continuation data:\n cont:". $continuation ."\n";
+			print " itct:". $itct ."\n";
+			print "\n";
+		}
+
+		# $nextUrl = 'https://www.youtube.com/browse_ajax?action_continuation=1&continuation='. $continuation;
+		$nextUrl = 'https://www.youtube.com/browse_ajax?ctoken='. $ctoken .'&continuation='. $continuation .'&itct='. $itct;
+	}
+
+    return (\@videos,$nextUrl);
+}
+
+sub fetch_playlist_json {
+    my $self = shift;
+    my $url = shift;
+
+    print 'Fetching playlist next page '. $url .' ... ' if $self->{verbose};
+
+    my $res = $self->ua->get($url,
+	# thanks youtube-dl for the hint towards these headers
+        'x-youtube-client-name' => '1',
+        'x-youtube-client-version' => '2.20200825.04.00', # version '1.20200609.04.02' will return a HTML blob in content_html, this version returns only JSON
+	);
+
+    croak('Error: fetch_playlist_json failed for url:'. $url .' status:'. $res->status_line) unless $res->is_success;
+
+    print "ok \n" if $self->{verbose};
+
+    return $res->decoded_content;
+}
+
+sub find_playlist_json_videos {
+	my $self = shift;
+	my $json = shift;
+
+	my $ref = JSON()->new->utf8(1)->decode($json);
+
+	## drill into YT's data structure
+	die "Error: YT JSON data not structed as expected!" unless
+	ref($ref) eq 'ARRAY' &&
+	$ref->[1] &&
+	$ref->[1]->{response} &&
+	$ref->[1]->{response}->{continuationContents} &&
+	$ref->[1]->{response}->{continuationContents}->{playlistVideoListContinuation} &&
+	$ref->[1]->{response}->{continuationContents}->{playlistVideoListContinuation}->{contents} && ref($ref->[1]->{response}->{continuationContents}->{playlistVideoListContinuation}->{contents}) eq 'ARRAY'
+	;
+
+	my $playlistVideoListContinuation = $ref->[1]->{response}->{continuationContents}->{playlistVideoListContinuation};
+
+	## weed out videos
+	my @videos;
+	my $cnt = 0;
+	for my $renderer (@{ $playlistVideoListContinuation->{contents} }){
+		$cnt++;
+		my $video = $renderer->{playlistVideoRenderer};
+
+		push(@videos, {
+			id	=> $video->{videoId},
+			runtime => $video->{lengthSeconds},
+			title	=> ($video->{title}->{simpleText}||'undef'), # encoded in Perl internal
+		});
+
+		if($self->{verbose}){
+			print " $cnt id:". $video->{videoId};
+			print " runtime:". ($video->{lengthSeconds}||'undef');
+			print " title:". Encode::encode_utf8($video->{title}->{simpleText}||'undef');
+			print "\n";
+		}
+	}
+
+	## next page
+	my $nextUrl;
+	if(	$playlistVideoListContinuation->{continuations} && ref($playlistVideoListContinuation->{continuations}) eq 'ARRAY' &&
+		$playlistVideoListContinuation->{continuations}->[0] &&
+		$playlistVideoListContinuation->{continuations}->[0]->{nextContinuationData}
```

Comment:
Couldn't we limit this to the final check? Maybe temporarily disable autovivification for this block of code?